src/OSPSuite.Core/Domain/ObjectPath.cs

```diff
@@ -211,20 +205,35 @@ IEnumerator IEnumerable.GetEnumerator()
       }
 
       /// <summary>
-      ///    Add one entry at the front of the path
+      ///    Add path entries to the end of the path
       /// </summary>
-      /// <param name="pathEntry">path entry to add</param>
-      public virtual void AddAtFront(string pathEntry)
+      public virtual void Add(string pathToAdd)
+      {
+         var pathElements = splitToElements(pathToAdd);
+         pathElements.Each(x => _pathEntries.Add(x));
+      }
+
+      /// <summary>
+      ///    Add path entries to the front of the path
+      /// </summary>
+      public virtual void AddAtFront(string pathToAdd)
+      {
+         var pathElements = splitToElements(pathToAdd);
+         pathElements.Reverse().Each(x => _pathEntries.Insert(0, x));
+      }
+
+      private static string[] splitToElements(string pathToInsert)
       {
-         _pathEntries.Insert(0, pathEntry);
+         var pathItems = pathToInsert.Split(new[] {PATH_DELIMITER}, StringSplitOptions.None);
+         return pathItems;
       }
 
       /// <summary>
-      ///   Add a multipart <paramref name="pathToAdd"/> at the front
+      ///    Add a multipart <paramref name="pathToAdd" /> at the front
       /// </summary>
       public virtual void AddAtFront(ObjectPath pathToAdd)
       {
-         pathToAdd.Reverse().Each(AddAtFront);
+         AddAtFront(pathToAdd.PathAsString);
```

Comment:
This is a bit weird. You have an array, you create a string to then create an array again. I think we need a lower level method tht takes an array of string